src/webview/leetCodeResultProvider.ts

```diff
@@ -34,19 +40,98 @@ class LeetCodeResultProvider implements Disposable {
         }
     }
 
-    private async provideHtmlContent(result: string): Promise<string> {
-        return `<!DOCTYPE html>
-            <html lang="en">
+    private parseResult(raw: string): Result {
+        try {
+            switch (raw[2]) {
+                case "√": {
+                    const result: AcceptResult = new AcceptResult();
+                    [result.status, raw] = raw.split(/  . (.+)([^]+)/).slice(1);
+                    [result.passed, raw] = raw.split(/\n  . (.+)([^]+)/).slice(1);
+                    [result.runtime, raw] = raw.split(/\n  . (.+)([^]+)/).slice(1);
+                    [result.memory] = raw.split(/\n  . (.+)/).slice(1);
+                    return result;
+                }
+                case "×": {
+                    const result: FailedResult = new FailedResult();
+                    [result.status, raw] = raw.split(/  . (.+)([^]+)/).slice(1);
+                    [result.passed, raw] = raw.split(/\n  . (.+)([^]+)/).slice(1);
+                    [result.testcase, raw] = raw.split(/\n  . testcase: '(.+)'([^]+)/).slice(1);
+                    [result.answer, raw] = raw.split(/\n  . answer: (.+)([^]+)/).slice(1);
+                    [result.expected, raw] = raw.split(/\n  . expected_answer: (.+)([^]+)/).slice(1);
+                    [result.stdout] = raw.split(/\n  . stdout: ([^]+?)\n$/).slice(1);
+                    result.testcase = result.testcase.replace("\\n", "\n");
+                    return result;
+                }
+                default: {
+                    throw new TypeError(raw);
+                }
+            }
+        } catch (error) {
+            leetCodeChannel.appendLine(`Result parsing failed: ${error.message}`);
+            throw error;
+        }
+    }
+
+    private getWebViewContent(result: Result): string {
+        const styles: string = this.mdEngine.getStylesHTML();
+        let body: string;
+        if (result instanceof AcceptResult) {
+            const accpet: AcceptResult = result as AcceptResult;
+            body = this.mdEngine.render([
+                `## ${result.status}`,
+                ``,
+                `* ${result.passed}`,
+                `* ${accpet.runtime}`,
+                `* ${accpet.memory}`,
+            ].join("\n"));
+        } else {
+            const failed: FailedResult = result as FailedResult;
+            body = this.mdEngine.render([
+                `## ${result.status}`,
+                `* ${result.passed}`,
+                ``,
+                `### Testcase`, // TODO: add command to copy raw testcase
```

Comment:
Curious, what will be the todo item here?

Reply:
Copy the raw testcase string for the convenience of submitting with custom testcase.

Reply:
Got it.